app/component/navbar.tsx

```diff
@@ -1,13 +1,91 @@
-import { useAccount, useNetwork } from "~app/storage"
+import { faCaretDown, faXmark } from "@fortawesome/free-solid-svg-icons"
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
+import { formatEther } from "ethers"
+import { useEffect, useState } from "react"
+
+import { useProviderContext } from "~app/context/provider"
+import { useAccount, useAccounts, useNetwork } from "~app/storage"
+import { type Account } from "~background/storage/local"
 import address from "~packages/util/address"
 
 export function Navbar() {
   const network = useNetwork()
   const account = useAccount()
+  const [isAccountModalOpened, setIsAccountModalOpened] = useState(false)
+  const toggleAccountModal = () =>
+    setIsAccountModalOpened(!isAccountModalOpened)
   return (
     <nav className="w-full grid grid-cols-5 justify-items-center my-4">
       <div>{network.chainId}</div>
-      <div className="col-span-3">{address.ellipsize(account.address)}</div>
+      <div className="col-span-3 cursor-pointer" onClick={toggleAccountModal}>
+        <span>{address.ellipsize(account.address)}</span>
+        <FontAwesomeIcon icon={faCaretDown} className="ml-2" />
+      </div>
+      {isAccountModalOpened && (
+        <AccountModal selected={account} onModalClosed={toggleAccountModal} />
+      )}
     </nav>
   )
 }
+
+function AccountModal(props: { selected: Account; onModalClosed: () => void }) {
+  const accounts = useAccounts()
+  return (
+    <div className="absolute top-0 left-0 w-screen h-screen p-4">
+      <div
+        className="absolute top-0 left-0 w-full h-full bg-black/75"
+        onClick={props.onModalClosed}
+      />
+      <div className="relative w-full p-4 bg-white rounded">
+        <div className="absolute top-4 right-4">
+          <button onClick={props.onModalClosed}>
+            <FontAwesomeIcon icon={faXmark} className="text-lg" />
+          </button>
+        </div>
+        {accounts.map((a) => (
+          <AccountPreview
+            account={a}
+            active={props.selected.address === a.address}
+          />
```

Comment:
Here we can use `a.address` as the `key` to avoid `Warning: Each child in a list should have a unique "key" prop.` errors.
```typescript
          <AccountPreview
            key={a.address}
            account={a}
            active={props.selected.address === a.address}
          />
```

Reply:
It's enough to use map index as key.